ls-ecaps.c

```diff
@@ -993,12 +1009,18 @@ dvsec_cxl_flex_bus(struct device *d, int where, int rev)
   if (rev > 1)
     printf(" 256BFlit%c PBRFlit%c",
         FLAG(w, PCI_CXL_FB_STAT_256B_FLIT), FLAG(w, PCI_CXL_FB_STAT_PBR_FLIT));
+  printf("\n");
 
-  l = get_conf_long(d, where + PCI_CXL_FB_MOD_TS_DATA);
-  data = BITS(l, 0, 24);
-  printf("\n\t\tFBModTS:\tReceived FB Data: %06x\n", (unsigned int)data);
+  // From Rev 1
+  if (rev >= 1)
+  {
+    l = get_conf_long(d, where + PCI_CXL_FB_MOD_TS_DATA);
+    data = BITS(l, 0, 24);
+    printf("\t\tFBModTS:\tReceived FB Data: %06x\n", (unsigned int)data);
+  }
```

Comment:
You are missing a check here that register `PCI_CXL_FB_MOD_TS_DATA` is really available, so to avoid reading data from uninitialized/unavailable config space. In previous suggest I wrote that for this it could make sense to move length check from the beginning of the function to this place, and call code for reading parsing this register conditionally. E.g.:

```suggestion
  if (rev >= 1 && len >= PCI_CXL_FB_MOD_TS_DATA + 4)
  {
    l = get_conf_long(d, where + PCI_CXL_FB_MOD_TS_DATA);
    data = BITS(l, 0, 24);
    printf("\t\tFBModTS:\tReceived FB Data: %06x\n", (unsigned int)data);
  }
```